src/vaadin-grid.html

```diff
@@ -445,6 +445,17 @@
         cell.appendChild(slot);
 
         cell._content = cellContent;
+        
+        cell.addEventListener('focus', () => {
+          this.dispatchEvent(new CustomEvent('cell-focus-changed', {
+            bubbles: true,
+            composed: true,
+            detail: {
+              cell: cell,
```

Comment:
The cell element is hidden inside the component's shadow root and shouldn't be accessed from outside. So either just dispatch the event directly on the cell element so that contextual information can be obtained with [getEventContext](https://github.com/vaadin/vaadin-grid/blob/master/src/vaadin-grid-event-context-mixin.html#L34) and/or fix the `detail` to only include the properties you'd get from `getEventContext`.

Reply:
That's a great idea, and it worked great. Much better than what I had. 

The only issue is that getting the actual content of the cell seems hard and I didn't find a straight-forward way of getting it. You can get the clicked column with `event.detail.column.path`, and you can get the item with `event.detail.item`. However you can't do event.detail.item[event.detail.column.path] as path might be "name.first". For that you should do `event.detail.item['name']['path]` or `item.name.path`. Leaving this up to the devs using our code will force them to write a lot of boilerplate and they will run into bugs later when they have taken only into account the single level path like 'firstname' and later add deep paths like 'address.city'.

Am I missing something? Is there an easy way to get the value of the cell from eventContext? is there a function in grid that takes item and column and returns the innerText? Should I add functionality to parse this from item and path, or should I still provide `cell._content.innerText` in the details?

Reply:
The problem with providing the inner text is that it's only a visual representation of the data and in practice can be anything. Especially if a developer assigns a renderer or a template to a column, inner text might might produce an unexpected value for the event details property.

Reply:
..for example if a column renderer that localizes the cell content is used, the value of `content` would depend on the user's locale

Reply:
Yes good point. Do you think getting the value from `items` would be good? Do you have a suggestion on how to do that from having the item and the column definition?

Reply:
At least Polymer has a helper for that `Polymer.Base.get(item, path)`

Reply:
`Polymer.Base.get(path, item)` is now in use.

Reply:
Hmm...now the `content` property is available only in events that occur on columns that define a `path`. That's kind of a strange limitation.

When users work with any other type of events, they need to get the context manually. Is there a reason why `cell-focus` should be an exception? To align this event, you'd have something like:
```js
grid.addEventListener('cell-focus', (event) => {
  const context = event.target.getEventContext(event);
  const content = Polymer.Base.get(context.column.path, context.item);
  output.textContent = content;
});
```
...where the event itself wouldn't automatically ship with any contextual data. This would also make it more lightweight – in most cases the data is not even needed. wdyt?